engine/src/test/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProviderTest.java

```diff
@@ -275,6 +286,42 @@ public void getAllGroupsCache() {
         assertEquals(updatedSize, users.size());
     }
 
+    /**
+     * Tests that calling MBOP for users works as expected.
+     */
+    @Test
+    void testGetUsersMBOP() {
+        try {
+            this.featureFlipper.setUseMBOPForFetchingUsers(true);
+
+            // Fake a REST call to MBOP.
+            final List<MBOPUser> MBOPUsers = this.mockGetMBOPUsers(this.MBOPMaxResultsPerPage);
+
+            final boolean adminsOnly = false;
+
+            // Return the mocked list first and then an empty one afterwards,
+            // since otherwise Mockito will always return the mocked list,
+            // which makes the MBOP loop go forever and hit heat space errors.
+            // In reality, an organization that has more than a 1000 users is
+            // going to be a rare case, and even if it is, the method can
+            // handle a few thousand elements.
+            Mockito
+                .when(this.mbopService.getUsersByOrgId(Mockito.eq(TestConstants.DEFAULT_ORG_ID), Mockito.eq(adminsOnly), Mockito.eq(RbacRecipientUsersProvider.MBOP_SORT_ORDER), Mockito.eq(this.MBOPMaxResultsPerPage), Mockito.anyInt()))
+                .thenReturn(MBOPUsers, new ArrayList<>());
```

Comment:
Could you test when the second "page" return some users?

Reply:
Yes! I've changed it so that we simulate two full pages plus and a third one with half a page, to test that the loop works correctly.

Reply:
Thanks!